speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl

```diff
@@ -80,3 +81,41 @@ DecisionMaker<SocketBlackBox, SDAConfig>::~DecisionMaker()
     delete m_recorder;
 #endif
 }
+
+template<typename SocketBlackBox, typename SDAConfig>
+void TEMP_DECISIONMAKER::RaceStop()
+{
+    std::string configPath("source-2.2.3\\data");
+    std::string configFile("database_connection_settings.txt");
+
+    if (!FindFileDirectory(configPath, configFile)) throw std::exception("Could not find database settings file");
+
+    std::ifstream ifstream(configPath + '\\' + configFile);
+    if (!ifstream.good()) throw std::exception("Could not open database settings file");
+
+    std::string ip;
+    std::string portString;
+    std::string username;
+    std::string password;
+    std::string schema;
+
+    try
+    {
+        ifstream >> ip;
+        ifstream >> portString;
+        ifstream >> username;
+        ifstream >> password;
+        ifstream >> schema;
+    }
+    catch (std::exception& e) { throw std::exception("Something went wrong while reading the config file");}
```

Comment:
You can check if opening an ifstream went correctly by doing if(ifstream.fail()) or if(ifstream.is_open()), don't use try and catches if you can avoid them

Reply:
That's not I'm checking here, but I can change it to check ifstream.eof() before every reading first